src/firebase/api/gcp-utils.js

```diff
@@ -1,74 +1,82 @@
+  userId, experimentId, image) => {
   const imagePath = `${userId}-${experimentId}/${image.name}`;
 
-  const storageRef = storage.ref();
-  const rawImageBucketRef = storageRef.child(bucket);
+  const rawImageBucketRef = app.storage(storageBuckets.SIE_RAW_IMGS).ref();
   const newImageRef = rawImageBucketRef.child(imagePath);
-  newImageRef.put(image).then((snapshot) => {
+  return await newImageRef.put(image).then(() => {
     return {
-      status: '201',
-      error: null,
+      status: 201,
+      message: 'image successfully uploaded',
+      data: {},
     };
-  }).catch((error) => {
+  }).catch((err) => {
     return {
-      status: '500',
-      data: null,
-      error,
+      status: 401,
+      message: `not authenticated ${err}`,
+      data: {},
     };
   });
 };
 
 
 /**
  * Subscribes to pubsub to signal stimuli generation completion
+ * @param {Function} imageUrlsHandler react hook function for imageUrls
+ * @param {Function} errorHandler react hook function for error.
  */
-export const observeStimuliCompletion = async () => {
-  gcpPubsub.topic(pubsub.SIE_RESULT).onPublish((msg) => {
-    // Decode the PubSub Message body.
-    const msgBody = msg.data ?
-      Buffer.from(message.data, 'base64').toString() : null;
-    // eslint-disable-next-line no-unused-vars
-    const participantId = msg.attributes.participant_id;
-    // eslint-disable-next-line no-unused-vars
-    const experimentId = msg.attributes.experiment_id;
+export const observeStimuliCompletion =
+  async (imageUrlsHandler, errorHandler) => {
+    gcpPubsub.topic(pubsub.SIE_RESULT).onPublish((msg) => {
+      // Decode the PubSub Message body.
+      const status = msg.data ?
+        Buffer.from(message.data, 'base64').toString() : null;
+      const userId = msg.attributes.participant_id;
+      const experimentId = msg.attributes.experiment_id;
 
-    if (msgBody === 'completed') {
-      // maybe this should be handled by a react hook?
-      // call getFileUrlsFromBucket once ready to display stimuli
-    } else {
-      // likewise: maybe should be handled by a setError hook
-    }
-  });
-};
+      if (status === 'Completed') {
+        // call getFileUrlsFromBucket once ready to display stimuli
+        getFileUrlsFromBucket(userId, experimentId).then(() => {
+          imageUrlsHandler(imageUrls);
+        });
+      } else {
+        // “face_missing”, “failed”
+        // call setError() for image processing status, display on frontend
+        errorHandler(status);
+      }
+    });
+  };
 
 /**
  * Get all file urls from bucket to display in img tags
- * @param {string} bucket storage bucket
- * @param {string} bucketPrefix prefix to destination storage directory
+ * @param {String} userId user id
+ * @param {String} experimentId experiment id
+ * @return {String[]} array of processed self image urls
  */
-export const getFileUrlsFromBucket = async (bucket, bucketPrefix) => {
+const getFileUrlsFromBucket = async (userId, experimentId) => {
```

Comment:
Is this function missing a return statement? I see URLS being pushed to the array, but the array is not being returned.

Reply:
It is in the end of the function which is auto folded I think.